### PR TITLE
Update bootstrap and docs for recent usage changes

### PR DESCRIPTION
**Pull request by CamStan**
### Description

This updates _bootstrap.sh_ to use a more recent and stable version of Argobots as the unstable versions will soon be removed ([related PR in Spack](https://github.com/spack/spack/pull/16458)).

This also applies a variety of documentation changes to bring the docs more up-to-date:

* build-intercept.rst
  - reorganizes the build documentation to make it easier to follow by fully separating the process of building with Spack vs building with Autotools. Puts them in separate drop-downs on the side menu.
  - update process for building with Spack
  - update process for building dependencies with Spack (GOTCHA version change, etc)
  - remove some redundency

* configuration.rst
  - updates the unifyfsd command line options

* dependencies.rst
  - updates the preferred version of Argobots
  - adds a link for the OpenSSL dependency

* examples.rst
  - update example program options
  - fix section names

* start-stop.rst
  - update example script for starting unifyfs servers
  - update unifyfs utilty command line options

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Testing (addition of new tests or update to current tests)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the UnifyFS code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted.

### CODE REVIEW

File: docs/start-stop.rst [hunk 1]
```diff
@@ -40,16 +40,15 @@ for further details on customizing the UnifyFS runtime configuration.
...
+    # spillover data to node-local ssd storage
+    export UNIFYFS_LOGIO_SPILL_DIR=/mnt/ssd/$USER/data
 
-        # store server logs in job-specific scratch area
-        export UNIFYFS_LOG_DIR=$JOBSCRATCH/logs
+    # store server logs in job-specific scratch area
+    export UNIFYFS_LOG_DIR=$JOBSCRATCH/logs
 
-        unifyfs start --share-dir=/path/to/shared/file/system &
+    unifyfs start --share-dir=/path/to/shared/file/system &
```
Comment (MichaelBrim) [inline docs/start-stop.rst:51]: Based on Hyogi's recent work, I suggest removing ampersand at the end of this line. There's no need to background the process anymore.

Comment (CamStan) [reply docs/start-stop.rst:51]: Ah, yup. Nice catch! I'll update that.

File: docs/build-intercept.rst [hunk 2]
```diff
@@ -62,41 +66,45 @@ your target system, it can be omitted during UnifyFS configuration by using
...
-Then install UnifyFS:
+Install UnifyFS
+^^^^^^^^^^^^^^^
 
 .. code-block:: Bash
 
     $ spack install unifyfs
     $ spack load unifyfs
 
+If the most recent changes on the develop branch of UnifyFS are desired, then do
```
Comment (MichaelBrim) [inline docs/build-intercept.rst:105]: rather than "develop branch", how about "development branch ('dev')"

File: docs/build-intercept.rst [hunk 3]
```diff
@@ -183,36 +194,36 @@ Then to manually build UnifyFS:
...
+<https://github.com/LLNL/UnifyFS/releases>`_ page or clone the develop branch
+from the `UnifyFS repository <https://github.com/LLNL/UnifyFS>`_.
 
-.. TODO: Add a section in build docs that shows all the build config options
+Build the Dependencies
+**********************
 
-Build the Dependencies without Spack
-"""""""""""""""""""""""""""""""""""""
+UnifyFS requires MPI(in progress of removing), LevelDB, GOTCHA, FlatCC, Margo
```
Comment (MichaelBrim) [inline docs/build-intercept.rst:213]: Until we no longer require MPI, we should just say we depend on it. Telling the users that we are in the process of removing that requirement doesn't really help them.

File: docs/build-intercept.rst (hunk 2 above)
Comment (CamStan) [reply docs/build-intercept.rst]: Good call. I mixed up which project calls the branch `dev` and which calls it `develop`.

File: docs/build-intercept.rst (hunk 3 above)
Comment (CamStan) [reply docs/build-intercept.rst]: Removed the parenthetical.

**Pull request merged**